### ISSUE TITLE
[Feature Request]: Add recipe to start a minimal environment for convenience of bug-reporting and testing

### ISSUE THREAD

**Issue by cireu** [emacs-lsp/lsp-mode#592]
English is not my mother-tongue, plz ignore my typo or grammarly issue ;)

I think it's useful to have a recipe (or Makefile target) like [this](https://github.com/abo-abo/swiper/blob/master/Makefile#L18) to help user start a minimal `lsp-mode` environment. It can help user locate the bug or issue more conveniently.

Since `lsp-mode` has more heavier dependencies than `ivy`. But I think we can let user to specific their package directory which includes all dependencies of `lsp-mode`. So we can using following code to add all the packages to `load-path`

```Elisp
(let ((default-directory (file-name-as-directory package-dir)))
  (push default-directory load-path)
  (normal-top-level-add-subdirs-to-load-path))
```

And now `lsp-mode` can require its dependencies and start a minimal environment.

**Comment by yyoncho** [comment]
I have a similar thing in `lsp-java` - there is quick start setup section and which then could be loaded `emacs -q -l quick-start.el` which will fire-up clean environment. I believe that this will be more suitable for `lsp-mode` than having make target since in `lsp-mode` the issue might be related to `lsp-ui`/`company-lsp` and the user might want to include/exclute components and it won't require cloning the repo. What do you think?

**Comment by cireu** [comment]
That's a good idea.

But I think it will be better for version control to put the recipe into a single file and tell user to run it like this
```Bash
emacs -q --eval '(with-current-buffer (url-retrieve-synchronously "https://raw.githubusercontent.com/emacs-lsp/lsp-mode/master/quick-start.el") (eval-buffer))'
```

**Comment by cireu** [comment]
And I also think it will be better to load these file with `(load "lsp-mode.el")` or `(require 'lsp-mode "lsp-mode.el")` rather than using `use-package`. Emacs prefer byte-compiled `.elc` file than `.el` file. But the backtrace from byte-compiled files are wired and not human-readable.

**Comment by cireu** [comment]
And I also think it will be better to load these file with `(load "lsp-mode.el")` or `(require 'lsp-mode "lsp-mode.el")` rather than using `use-package`. Emacs prefer byte-compiled `.elc` file than `.el` file. But the backtrace from byte-compiled files are wired and not human-readable.

**Comment by yyoncho** [comment]
I believe that we may force loading of `.el` files by modifying `load-suffixes`.

I would like to use `use-package` since this configuration could be used for multiple purposes - a quick start and also as a minimal environment.

**Comment by cireu** [comment]
Oh, my bad. I never noticed it's possible to change the loading action of Emacs by setting the `load-suffix`.

So it's OK to use `use-package`

I wrote a simple script like this.
```Elisp
(let ((load-suffixes '(".el"))
      (package-archives '("melpa" . "http://melpa.org/packages/"))
      (use-package-always-ensure t)
      (use-package-expand-minimally t))
  (unless (require 'use-package nil t)
    (package-initialize)
    (package-refresh-contents)
    (package-install 'use-package)
    (require 'use-package))
  (use-package lsp-mode)
  (use-package lsp-ui)
  (use-package company-lsp))
```

**Comment by yyoncho** [comment]
Do you want to provide a pr?

**Comment by cireu** [comment]
I'm going to sleep now… maybe I can do this tomorrow morning ;)

**Comment by cireu** [comment]
Oops! I meet problem.

After run
```
(let ((package-archives '(("melpa" . "http://elpa.emacs-china.org/melpa/")
                          ("gnu" . "http://elpa.emacs-china.org/gnu/")))
      (use-package-always-ensure t)
      (use-package-expand-minimally t)
      (use-package-always-demand t))
  (setq debug-on-error t)
  (unless (require 'use-package nil t)
    (require 'package)
    (package-initialize)
    (package-refresh-contents)
    (package-install 'use-package)
    (require 'use-package))
  (let ((load-suffixes (remove ".elc" load-suffixes)))
    (use-package lsp-mode)
    (use-package lsp-ui)
    (use-package company-lsp)))
```

It failed and I got message 
```
insert-file-contents: Recursive load: "/usr/share/emacs/26.1/lisp/jka-compr.el.gz", "/usr/share/emacs/26.1/lisp/jka-compr.el.gz", "/usr/share/emacs/26.1/lisp/jka-compr.el.gz", "/usr/share/emacs/26.1/lisp/jka-compr.el.gz", "/usr/share/emacs/26.1/lisp/jka-compr.el.gz", "/usr/share/emacs/26.1/lisp/emacs-lisp/warnings.el.gz"
```

**Comment by yyoncho** [comment]
Fixed by #594
